packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const { hasRWEnabled } = require('../../utils/review-workflows');
+const { getService } = require('../../utils');
+
+/**
+ * Decorates the entity service with RW business logic
+ * @param {object} service - entity service
+ */
+const decorator = (service) => ({
+  async create(uid, opts = {}) {
+    const model = strapi.getModel(uid);
+    const hasRW = hasRWEnabled(model);
+
+    const entity = await service.create.call(this, uid, opts);
+    if (!hasRW) {
+      return entity;
+    }
+
+    // Assign this entity to the default workflow stage
+    const { assignEntityDefaultStage } = getService('review-workflows');
+    await assignEntityDefaultStage(uid, entity.id);
+
+    return entity;
+  },
```

Comment:
If I understand correctly the entity is created and then assigned the default workflow if it's not assigned no?

I see a couple of issues/inconveniences with this approach:
- The triggered webhooks and lifecycles events in the `service.create.call` will not have the stage information.
- We are making additional database queries if RW is enabled.

I can see this minor problems fixed if we do something similar as i18n:
```js
  // packages/plugins/i18n/server/services/entity-service-decorator.js
  async create(uid, opts = {}) {
    const model = strapi.getModel(uid);

    const { syncLocalizations, syncNonLocalizedAttributes } = getService('localizations');
    const { isLocalizedContentType } = getService('content-types');

    if (!isLocalizedContentType(model)) {
      return service.create.call(this, uid, opts);
    }

    const { data } = opts;
    await assignValidLocale(data); // <==== We assign the locale directly in the parameters if it is not set

    const entry = await service.create.call(this, uid, opts);
	...
    return entry;
  },
  ```

Do you think this could be possible here?